### PR TITLE
Updating `libsndfile` to version 1.0.31 to fix CVEs.

### PR DESCRIPTION
**Pull request by PawelWMS**
Updated `libsndfile` to fix the following CVEs:

-  2017-6892, 2017-8361, 2017-8362, 2017-8363, 2017-8365, 2017-12562, 2017-14245, 2017-14246, 2017-14634,
-  2018-13139, 2018-19432, 2018-19661, 2018-19662, 2018-19758,
-  2019-3832.

Also:
- Updated other patches to work with new source.
- Removed CVE patches, which were no longer necessary (new version covers them).
- Updated the installed files' list.
- Nopatched a disputed CVE.

### CODE REVIEW

File: SPECS/libsndfile/libsndfile.spec [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 URL:            http://www.mega-nerd.com/libsndfile/
-Source0:        http://www.mega-nerd.com/libsndfile/files/libsndfile-%{version}.tar.gz
+#Source0:        https://github.com/libsndfile/libsndfile/archive/%{version}.tar.gz
+Source0:        %{name}-%{version}.tar.gz
 
-Patch0:         libsndfile-1.0.25-system-gsm.patch
+Patch0:         libsndfile-1.0.31-system-gsm.patch
 Patch1:         libsndfile-1.0.25-zerodivfix.patch
```
Comment (jslobodzian) [inline SPECS/libsndfile/libsndfile.spec:13]: > libsndfile-1.0.25-zerodivfix.patch [](start = 15, length = 35)

should this be removed too?

**Review by jslobodzian — APPROVED** [review]
:shipit:

File: SPECS/libsndfile/libsndfile.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/libsndfile/libsndfile.spec:13]: No, this is one of the patches applied by Fedora and a few other distros unrelated to any CVE known to me. I didn't want to remove it and risk any run-time or security issues.

Comment (PawelWMS) [reply SPECS/libsndfile/libsndfile.spec:13]: Same goes for the patch below it.

**Pull request merged**